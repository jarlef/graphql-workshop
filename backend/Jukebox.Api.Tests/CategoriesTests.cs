using Jukebox.Api.Tests.Infrastructure;
using Jukebox.Data;
using Snapshooter.Xunit;

namespace Jukebox.Api.Tests;

public class CategoriesTests(JukeboxApplicationFactory factory)
    : BaseIntegrationTest(factory),
        IAsyncLifetime
{
    [Fact]
    public async Task GetCategories_ShouldReturnAllCategories()
    {
        var result = await Client.GetCategories.ExecuteAsync();
        result.Data.MatchSnapshot();
    }

    public async Task InitializeAsync()
    {
        DbContext.Genre.Add(new Genre { Name = "Foo" });
        DbContext.Genre.Add(new Genre { Name = "Bar" });
        await DbContext.SaveChangesAsync();
    }

    public Task DisposeAsync()
    {
        return Task.CompletedTask;
    }
}
