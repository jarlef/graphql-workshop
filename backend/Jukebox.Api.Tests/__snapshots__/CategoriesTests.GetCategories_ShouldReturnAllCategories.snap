{
  "Categories": [
    {
      "Id": 1,
      "Name": "Foo"
    },
    {
      "Id": 2,
      "Name": "Bar"
    }
  ]
}
